tests/parser.cpp

```diff
@@ -1971,6 +1971,121 @@ TEST(Parser, invalid_provider)
   test_parse_failure("asdf:xyz:www { }");
 }
 
+TEST(Parser, subprog_only)
+{
+  test("fn f1(): void {}",
+       "Program\n"
+       " f1: void\n");
```

Comment:
Nit: would it be possible to add `()` here so that it is visible that it's a function?
```suggestion
       " f1: void()\n");
```